File: src/main/java/com/core/base/corebase/domain/review/ReviewMember.java
package com.core.base.corebase.domain.review;

import com.core.base.corebase.domain.user.Member;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "review_member")
public class ReviewMember {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "review_id", nullable = false)
    private Review review;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    protected ReviewMember() {
        // For JPA
    }

    public ReviewMember(Member member, Review review) {
        this.member = member;
        this.review = review;
    }

}

File: src/main/kotlin/com/core/base/corebase/domain/review/ReviewMember.kt
package com.core.base.corebase.domain.review

import com.core.base.corebase.domain.user.Member
import jakarta.persistence.*

@Entity(name = "review_member")
class ReviewMember(
    member: Member, //Member
    review: Review, // Review
){

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0L


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "review_id", nullable = false)
    var review: Review = review; protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    var member: Member = member; protected set

}